tests/test_plugins/test_adjoint.py

```diff
@@ -1913,6 +1930,10 @@ def test_sidewall_angle_validator(log_capture, sidewall_angle, log_expected):  #
 
     jax_polyslab1 = JaxPolySlab(axis=POLYSLAB_AXIS, vertices=VERTICES, slab_bounds=(-1, 1))
 
+    msg = "'JaxPolySlab' does not yet perform the full adjoint gradient treatment "
+    for entry in td.log._static_cache:
```

Comment:
we have an `assert_log_level` and `AssertLogLevel` (context manager version) in `tests/utils`. But this is probably fine.